reducer/field/count.go

```diff
@@ -24,7 +23,7 @@ func (c *Count) Consume(v zeek.Value) error {
 	var i zeek.Int
 	//XXX need CoerceToCount?
 	if !zeek.CoerceToInt(v, &i) {
-		return zng.ErrTypeMismatch
+		return errCoerce
 	}
```

Comment:
Seems like errCoerce should be exportable.  And I would again question why isn't this a type mismatch?

Reply:
Is this error even exported?  It's a way for the reducers to track field values that were encountered that cannot be reduced according to the function applied.  It seems like ErrTypeMismatch is the right error here.

Reply:
Reverted. Now I've worked on this a little, I think what matters most is properly wrapping errors to give context on where an error happened. This PR does this for record typechecks; others remain to be done in follow-on PRs.